tests/integration/snapshot/v3_snapshot_test.go

```diff
@@ -284,11 +283,16 @@ func restoreCluster(t *testing.T, clusterN int, dbPath string) (
 }
 
 // TODO: TLS
-func newEmbedURLs(n int) (urls []url.URL) {
+func newEmbedURLs(t testutil.TB, n int) (urls []url.URL) {
 	urls = make([]url.URL, n)
 	for i := 0; i < n; i++ {
-		rand.Seed(int64(time.Now().Nanosecond()))
-		u, _ := url.Parse(fmt.Sprintf("unix://localhost:%d", rand.Intn(45000)))
+		l := integration2.NewLocalListener(t)
```

Comment:
Why change this code?

Reply:
this change is not there in my local commit, i was wondering too how this file got changed.

Reply:
I expect an incorrect local rebase.

Reply:
yes, i did a rebase before i pushed the commit.

after doing the changes you suggested, i did a rebase again to main with `git rebase main` (after updating main), but after rebase the commit seem to have vanished. 

What is the practice followed in etcd? feature  branches should have a linear history so that PR merge is easier. but when i rebase, the commit gets lost